### ISSUE TITLE
MagicDNS only works partially

### ISSUE THREAD

**Issue by sylr** [tailscale/tailscale#884]
**Describe the bug**

I have deployed Tailscale 1.2.0 on a few nodes in order to test MagicDNS and got it working only partially.

I've setup a coredns server on one of the nodes which is working fine when used with the tailscale IP but as soon as I enable MagiDNS my nodes do not resolve anymore except for the "tailscale names", e.g.: `mynode.mydomain.beta.tailscale.net.`.

Any other resolution is "lost". I see the source node sending the DNS requests to 100.100.100.100 but they are not arriving on the DNS server configured in the tailscale admin page.

It looks like 100.100.100.100 is not forwarding the DNS requests.

**To Reproduce**

**Expected behavior**

**Screenshots**

**Version information:**
 - Device: Raspberry
 - OS: Raspbian
 - OS version: Raspbian GNU/Linux 10 (buster)
 - Tailscale version: 1.2.0

**Additional context**

**Comment by veera83372** [comment]
Same problem without enabling magic DNS too,

My setup is as follows:

LAN - 192.168.4.1/24
gateway - 192.168.4.1 (rapberrypi), 100.x.x.x(tailscale0)
I am running `dnsmasq` in my gateway.
Whenever I add Gateways's tailscale ip as dns server, all my public dns queries are not responding. 

Adding comment here because it is problem with DNS than magicDNS

```
$ tailscale version
1.2.6
  tailscale commit: 3f4e6d959a246369dab5866e21a2f7a0999cf715
  other commit: aad952c9af93ebb1d54a5ff2aee05190cac5d474
  go version: go1.15.4-tsf9db43b
```

**Comment by danderson** [comment]
Should now be fixed, and will roll out in Tailscale 1.8. You can test-drive things with the unstable package on linux, or building from source on macOS/windows. We're still working on applying the polish, so you might hit rough edges, but the situation you ran into should no longer occur.
